modules/channel-messenger/src/backend/messenger.ts

```diff
@@ -307,13 +322,19 @@ export class MessengerClient {
     await this._callEndpoint('/messages', body)
   }
 
-  async sendTextMessage(senderId: string, message: string) {
-    const body = {
-      recipient: {
-        id: senderId
-      },
-      message
-    }
+  /**
+   * @see https://developers.facebook.com/docs/messenger-platform/reference/send-api
+   */
+  async sendTextMessage(senderId: string, message: any) {
```

Comment:
A better solution would be to restrict the type of `message` to `MessengerTextMessage | MessengerAttachmentMessage` instead of `any`. This will remove the need for the `okKeys.reduce()` statement below.

Reply:
Thanks for your review @spg 

Unfortunately, it won’t remove the need for the okKeys.reduce(), since typescript doesn’t typecheck at runtime. In fact, as of today, the signature is `async sendTextMessage(senderId: string, message: string) `, but it’s not a string that is passed to the `sendTextMessage` method for the `message` argument) and typescript does not catch the mismatch.

Plus, the goal of my change is to ensure that a valid messenger’s message object is sent to messenger (for now, the object passed as argument to sendTextMessage may not be a messenger valid object, since it may have some extra fields like `agent` and `preview`), that’s why I use the `any` type.

After some extra thoughts, it feels weird to expect a pre-formatted object ready to be sent to messenger here. (I reckon the builtin module is doing the normalization : https://github.com/botpress/botpress/tree/master/modules/builtin/src/content-types). Doesn’t this eventually break compatibility with other modules (like the `hitl` module, that does not expect à messenger pre-formatted object)?

Reply:
Hey @gorghoa 

I just checked out your code, and it seems like the `body` variable is missing:

![Screen Shot 2020-02-14 at 09 03 59](https://user-images.githubusercontent.com/892367/74538097-74bc0300-4f09-11ea-855b-cb851e4ad96d.png)

This error should have been caught by the compiler. Any ides why this error was not caught @allardy ?

Reply:
@spg Nop, errors are correctly handled in views, but they aren't in the backend. I haven't found the issue yet :/

Reply:
Where are we with those changes guys ? Did we agree on something and is this ready to merge ?

Reply:
I solved the issue raised by @spg , but otherwise I don't have an opinion on this, I don't have a messenger bot to test it

Reply:
@spg, 

> I just checked out your code, and it seems like the body variable is missing

Yes, you’re  right. I did the renaming too quickly, thanks :+1: